crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -241,12 +395,16 @@ impl CachingNativePriceEstimator {
         update_size: Option<usize>,
         prefetch_time: Duration,
         concurrent_requests: usize,
+        fetcher: Option<Arc<dyn NativePriceBatchFetcher>>,
     ) -> Self {
+        let cache: Arc<TokioMutex<HashMap<H160, CachedResult>>> = Default::default();
         let inner = Arc::new(Inner {
             estimator,
-            cache: Default::default(),
+            cache: cache.clone(),
             high_priority: Default::default(),
             max_age,
+            batching: fetcher
+                .map(|fetcher| Batching::new(cache.clone(), Duration::from_millis(50), fetcher)),
```

Comment:
The configuration of 50 ms is hardcoded, but it will be changed in the upcoming PR (2/2). It has NO effect, because in order to enable this, `fetcher` has to be `Some`, which is NEVER the case.